File: Algorithms/083.RemoveDuplicatesFromSortedList.java
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) {
            return head;
        }
        ListNode p = head.next;
        ListNode previous = head;
        while (p != null) {
            if (p.val == previous.val) {
                previous.next = p.next;
            }
            else {
                previous = previous.next;
            }
            p = p.next;
        }
        return head;
    }
}

File: Algorithms/py/083.RemoveDuplicatesFromSortedList.py
class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if not head:
            return head
        previous = head
        current = head.next
        while current:
            if current.val == previous.val:
                previous.next = current.next
                tmp = current
                current = current.next
                del tmp
            else:
                previous = previous.next
                current = current.next
        return head
